ui/views.py

```diff
@@ -210,11 +213,15 @@ class CollectionViewSet(viewsets.ModelViewSet):
 
     def get_queryset(self):
         """
-        Custom get_queryset to filter collections that user has admin access to.
+        Custom get_queryset to filter collections.
+        If no key is requested, query for collections the user has admin access to.
+        If a key is requested, query for collections the user has view access to.
         """
         if self.request.user.is_superuser:
             return Collection.objects.all()
-        return Collection.objects.all_admin(self.request.user)
+        if self.kwargs.get('key'):  # Detail view
```

Comment:
A few things:
- In #137 I removed the 2 lines above because they're redundant (check `Collection.objects.all_admin` implementation). You can do the same here.
- I think I'd rather have the docstring comments in the body of the function. This makes the comments more proximal to the code, and it is approx. 1 billion times more likely that someone will forget to update the docstring after an implementation change than it is for them to forget to update an inline comment.
- Personal preference, but I'd rather not have comments at the end of any line of code. I prefer seeing them above the relevant line of code, and we seem to follow that pattern here generally.

Reply:
OK